tests/test_training/test_opt.py

```diff
@@ -47,6 +48,9 @@ def test_S2gate_coincidence_prob(n):
     skip_np()
 
     settings.SEED = 40
+    rng = tf.random.get_global_generator()
+    rng.reset_from_seed(settings.SEED)
```

Comment:
should this logic go into the `SEED` setter?

Reply:
agreed, but that's probably out of scope for this PR so I suppose we can leave it for now.